core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackendUtil.scala

```diff
@@ -0,0 +1,117 @@
+ */
+private[spark] object MesosSchedulerBackendUtil extends Logging {
+  /**
+   * Parse a volume spec in the form passed to 'docker run -v'
+   * which is [host-dir:][container-dir][:rw|ro]
+   */
+  def parseVolumesSpec(volumes: String): List[Volume] = {
+    volumes.split(",").map(_.split(":")).map { spec =>
+        val vol: Volume.Builder = Volume
+          .newBuilder()
+          .setMode(Volume.Mode.RW)
+        spec match {
+          case Array(container_path) => 
+            Some(vol.setContainerPath(container_path))
+          case Array(container_path, "rw") =>
+            Some(vol.setContainerPath(container_path))
+          case Array(container_path, "ro") =>
+            Some(vol.setContainerPath(container_path)
+                    .setMode(Volume.Mode.RO))
+          case Array(host_path, container_path) => 
+            Some(vol.setContainerPath(container_path)
+                    .setHostPath(host_path))
+          case Array(host_path, container_path, "rw") =>
+            Some(vol.setContainerPath(container_path)
+                    .setHostPath(host_path))
+          case Array(host_path, container_path, "ro") =>
+            Some(vol.setContainerPath(container_path)
+                    .setHostPath(host_path)
+                    .setMode(Volume.Mode.RO))
+          case spec => {
+            logWarning("parseVolumeSpec: unparseable: " + spec.mkString(":"))
+            None
+          }
+      }
+    }
+    .flatMap { _.map(_.build) }
+    .toList
+  }
+
+  /**
+   * Parse a portmap spec, similar to the form passed to 'docker run -p'
+   * the form accepted is host_port:container_port[:proto]
+   * Note:
+   * the docker form is [ip:]host_port:container_port, but the DockerInfo
+   * message has no field for 'ip', and instead has a 'protocol' field.
+   * Docker itself only appears to support TCP, so this alternative form
+   * anticipates the expansion of the docker form to allow for a protocol
+   * and leaves open the chance for mesos to begin to accept an 'ip' field
+   */
+  def parsePortMappingsSpec(portmaps: String): List[DockerInfo.PortMapping] = {
+    portmaps.split(",").map(_.split(":")).map { spec: Array[String] =>
+      val portmap: DockerInfo.PortMapping.Builder = DockerInfo.PortMapping
+        .newBuilder()
+        .setProtocol("tcp")
+      spec match {
+        case Array(host_port, container_port) =>
+          Some(portmap.setHostPort(host_port.toInt)
+                      .setContainerPort(container_port.toInt))
+        case Array(host_port, container_port, protocol) =>
+          Some(portmap.setHostPort(host_port.toInt)
+                      .setContainerPort(container_port.toInt)
+                      .setProtocol(protocol))
+        case spec => {
+          logWarning("parsePortMappingSpec: unparseable: " + spec.mkString(":"))
+          None
+        }
+      }
+    }
+    .flatMap { _.map(_.build) }
+    .toList
+  }
+
+  def withDockerInfo(
```

Comment:
This method name is still not super descriptive. Should this be `setupDockerContainer`? Also please add a short javadoc on what this does.